test/cypress/grid/integration/resize.cy.js

```diff
@@ -0,0 +1,27 @@
+describe('Grid resize.', () => {
+    beforeEach(() => {
+        cy.visit('dashboards/cypress/component-grid-resize');
+    });
+
+    it('Changes in the grid should be reflected in the options', () => {
+        cy.get('.highcharts-datagrid-header-cell').eq(0).as('cell');
+        cy.get('@cell').find('.highcharts-datagrid-column-resizer').as('resizer');
+        cy.get('@resizer').trigger('mousedown');
+        cy.get('@resizer').trigger('mousemove', { pageX: 100 });
+        cy.get('@resizer').trigger('mouseup');
+
+        cy.get('@cell').invoke('width').should('be.lessThan', 400);
+
+        cy.board().then((board) => {
+            const options = board.getOptions().components[0];
```

Comment:
1. Actually it's `componentOptions` not options
2. IMO we should extract options from `mountedComponents`, not components.